C/services/filter-plugin-interfaces/python/python_plugin_interface.cpp

```diff
@@ -0,0 +1,435 @@
+        PyList_Append(sysPath, PyUnicode_FromString((char *) foglampPythonDir.c_str()));
+
+        // Set sys.argv for embedded Python 3.5
+        int argc = 2;
+        wchar_t* argv[2];
+        argv[0] = Py_DecodeLocale("", NULL);
+        argv[1] = Py_DecodeLocale(pluginName, NULL);
+        PySys_SetArgv(argc, argv);
+
+        // 2) Import Python script
+        pModule = PyImport_ImportModule(name.c_str());
+
+        // Check whether the Python module has been imported
+        if (!pModule)
+        {
+                // Failure
+                if (PyErr_Occurred())
+                {
+                        logErrorMessage();
+                }
+                Logger::getLogger()->fatal("PluginInterfaceInit: cannot import Python 3.5 script "
+                                           "'%s' from '%s' : pythonScript=%s, shimLayerPath=%s, plugin '%s'",
+                                           name.c_str(), path.c_str(),
+                                           pythonScript.c_str(),
+                                           shimLayerPath.c_str(),
+                                           sPluginName.c_str());
+        }
+        else
+        {
+                Logger::getLogger()->debug("%s:%d: python module loaded successfully, pModule=%p, plugin '%s'",
+                                           __FUNCTION__,
+                                           __LINE__,
+                                           pModule,
+                                           sPluginName.c_str());
+        }
+
+        PyGILState_Release(state);
+
+        return pModule;
+}
+
+/**
+ * Returns function pointer that can be invoked to call '_sym' function
+ * in python plugin
+ */
+void* PluginInterfaceResolveSymbol(const char *_sym)
+{
+	string sym(_sym);
+	if (!sym.compare("plugin_info"))
+		return (void *) plugin_info_fn;
+	else if (!sym.compare("plugin_init"))
+		return (void *) filter_plugin_init_fn;
+	else if (!sym.compare("plugin_shutdown"))
+		return (void *) plugin_shutdown_fn;
+	else if (!sym.compare("plugin_reconfigure"))
+		return (void *) plugin_reconfigure_fn;
+	else if (!sym.compare("plugin_ingest"))
+		return (void *) filter_plugin_ingest_fn;
+	else if (!sym.compare("plugin_start"))
+	{
+		Logger::getLogger()->warn("FilterPluginInterface currently "
+					  "does not support 'plugin_start'");
+		return NULL;
+	}
+	else
+	{
+		Logger::getLogger()->fatal("FilterPluginInterfaceResolveSymbol can not find symbol '%s' "
+					   "in the Filter Python plugin interface library, "
+					   "loaded plugin '%s'",
+					   _sym,
+					   sPluginName.c_str());
+		return NULL;
+	}
+}
+
+/**
+ * Create a list od dict Python object (PyList) from
+ * a vector of Readind pointers
+ *
+ * @param    readings   The input readings vector
+ * @return              PyList object on success or NULL on errors
+ */
+PyObject* createReadingsList(const vector<Reading *>& readings)
```

Comment:
Can you move this function to C/services/common-plugin-interfaces/python/pyobject_reading_parser.cpp, it might be used by some other code also for sending readings to python side.